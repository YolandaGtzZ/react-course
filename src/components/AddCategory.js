import React, { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setinputValue] = useState("");

  const handleInputChange = (e) => {
    setinputValue(e.target.value);

    console.log("HandleInputChange llamado");
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('handleSubmit');

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setinputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <div>
        <input type="text" value={inputValue} onChange={handleInputChange} />
      </div>
    </form>
  );
};

AddCategory.protoTypes = {
  setCategories: PropTypes.func.isRequired,
};
