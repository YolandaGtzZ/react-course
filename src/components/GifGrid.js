import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

  useFetchGifs.mockReturnValue({
    data:[ ],
    loading: true
  });

  const { data:images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className='animate__animated animate__swing'>{category}</h3>
      {loading && <p>Loading</p>}
      <div >
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};


GifGrid.propTypes ={
  category: PropTypes.string.isRequired
};

