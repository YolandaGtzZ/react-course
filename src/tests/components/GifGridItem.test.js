import React from "react";
import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe("Debe de mostrar el componente <GifGridItem />", () => {
  const title = "un titulo";
  const url = "https://localhost.com";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("debe demostrasrlo correctamente", () => {    
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de tener un parrafo con el title', () => {
     
    const p =wrapper.find('p');
    expect(p.text().trim()).toBe(title);
  });

  test('debe de tener la img igual a los props', () => {
      const img = wrapper.find('img');
      console.log(img.prop());     
  })  
  
});
