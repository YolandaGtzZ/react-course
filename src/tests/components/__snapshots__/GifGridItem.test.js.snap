// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debe de mostrar el componente <GifGridItem /> debe demostrasrlo correctamente 1`] = `
<div>
  <img
    alt="un titulo"
    src="https://localhost.com"
  />
  <p>
    un titulo
  </p>
</div>
`;
